### PR TITLE
Add tag for dual-channel logs

### LINKED ISSUE #986: Add [Dual Channel] Tag for Dual Channel Server Logs
For every server log entry that is related to the Dual Channel Replication functionality, add one of the following tags at the beginning of the log message:

- [Dual Channel Replication]
- \<Dual Channel Replication>
- [Dual Channel]
- \<Dual Channel>
- [DCR]
- \<DCR>
- [Dual Channel Sync]
- \<Dual Channel Sync>

This will help easily identify and filter logs related to the Dual Channel Replication operations, improving readability and troubleshooting.
discused here https://github.com/valkey-io/valkey/pull/945/files#r1740179518

### PR DESCRIPTION
**Pull request by naglera**
This PR introduces a consistent tagging system for dual-channel logs. The goal is to improve log readability and filterability, making it easier for operators to manage and analyze log entries.

Resolves https://github.com/valkey-io/valkey/issues/986

### CODE REVIEW

**Review by hwware — APPROVED** [review]

**Review by ranshid — COMMENTED** [review]
Can we consider define a new DualChannel logger?

Like:
```
#define DualChannelServerLog(level, ...) \
    serverLog(level, "<Dual Channel> " __VA_ARGS__)
```

IMO it would make the logging and code cleaner

File: src/replication.c [hunk 1]
```diff
@@ -2607,7 +2610,8 @@ static void fullSyncWithPrimary(connection *conn) {
             argc++;
             err = sendCommandArgv(conn, argc, args, lens);
             if (err) {
-                serverLog(LL_WARNING, "Sending command to primary in dual channel replication handshake: %s", err);
+                serverLog(LL_WARNING,
```
Comment (ranshid) [inline src/replication.c]: why not use the macro here?

File: src/replication.c [hunk 2]
```diff
@@ -2652,17 +2657,19 @@ static void fullSyncWithPrimary(connection *conn) {
...
         err = receiveSynchronousResponse(conn);
         if (err == NULL) {
-            serverLog(LL_WARNING, "Primary did not respond to replconf command during SYNC handshake");
+            dualChannelServerLog(LL_WARNING, "Primary did not respond to replconf command during SYNC handshake");
             goto error;
         }
 
         if (err[0] == '-') {
             serverLog(LL_NOTICE,
-                      "Server does not support sync with offset, dual channel sync approach cannot be used: %s", err);
```
Comment (ranshid) [inline src/replication.c]: why not replace with the macro here?

File: src/replication.c [hunk 3]
```diff
@@ -2772,13 +2779,13 @@ int readIntoReplDataBlock(connection *conn, replDataBufBlock *data_block, size_t
...
         if (connGetState(conn) != CONN_STATE_CONNECTED) {
-            serverLog(LL_NOTICE, "Error reading from primary: %s", connGetLastError(conn));
+            dualChannelServerLog(LL_NOTICE, "Error reading from primary: %s", connGetLastError(conn));
             cancelReplicationHandshake(1);
         }
         return C_ERR;
     }
     if (nread == 0) {
-        serverLog(LL_VERBOSE, "Provisional primary closed connection");
+        serverLog(LL_VERBOSE, "<Dual Channel> Provisional primary closed connection");
```
Comment (ranshid) [inline src/replication.c]: macro?

File: src/replication.c [hunk 4]
```diff
@@ -3009,8 +3016,9 @@ int replicaTryPartialResynchronization(connection *conn, int read_reply) {
             /* While in dual channel replication, we should use our prepared repl id and offset. */
             psync_replid = server.repl_provisional_primary.replid;
             snprintf(psync_offset, sizeof(psync_offset), "%lld", server.repl_provisional_primary.reploff + 1);
-            serverLog(LL_NOTICE, "Trying a partial resynchronization using main channel (request %s:%s).", psync_replid,
-                      psync_offset);
+            serverLog(LL_NOTICE,
```
Comment (ranshid) [inline src/replication.c]: macro?

File: src/networking.c [hunk 5]
```diff
@@ -1923,14 +1923,15 @@ int freeClientsInAsyncFreeQueue(void) {
             if (!c->rdb_client_disconnect_time) {
                 if (c->conn) connSetReadHandler(c->conn, NULL);
                 c->rdb_client_disconnect_time = server.unixtime;
-                serverLog(LL_VERBOSE, "Postpone RDB client id=%llu (%s) free for %d seconds", (unsigned long long)c->id,
-                          replicationGetReplicaName(c), server.wait_before_rdb_client_free);
+                serverLog(LL_VERBOSE, "<Dual Channel> Postpone RDB client id=%llu (%s) free for %d seconds",
```
Comment (ranshid) [inline src/networking.c]: why not use the macro?

File: src/networking.c [hunk 6]
```diff
@@ -1682,7 +1682,7 @@ void freeClient(client *c) {
     /* Log link disconnection with replica */
     if (getClientType(c) == CLIENT_TYPE_REPLICA) {
         serverLog(LL_NOTICE,
-                  c->flag.repl_rdb_channel ? "Replica %s rdb channel disconnected."
+                  c->flag.repl_rdb_channel ? "<Dual Channel> Replica %s rdb channel disconnected."
```
Comment (ranshid) [inline src/networking.c]: I see no harm in moving the check higher and use the macro in this case.

File: src/networking.c [hunk 7]
```diff
@@ -1923,14 +1923,15 @@ int freeClientsInAsyncFreeQueue(void) {
...
-                          replicationGetReplicaName(c), server.wait_before_rdb_client_free);
+                serverLog(LL_VERBOSE, "<Dual Channel> Postpone RDB client id=%llu (%s) free for %d seconds",
+                          (unsigned long long)c->id, replicationGetReplicaName(c), server.wait_before_rdb_client_free);
             }
             if (server.unixtime - c->rdb_client_disconnect_time <= server.wait_before_rdb_client_free) continue;
-            serverLog(LL_NOTICE,
-                      "Replica main channel failed to establish PSYNC within the grace period (%ld seconds). "
-                      "Freeing RDB client %llu.",
-                      (long int)(server.unixtime - c->rdb_client_disconnect_time), (unsigned long long)c->id);
+            serverLog(
```
Comment (ranshid) [inline src/networking.c]: why not use the macro?

**Review by ranshid — COMMENTED** [review]
Some places I wonder why we havn't used the macro

**Review by ranshid — APPROVED** [review]
I think this is a safe change and the failed test seems to be related to something else.

**Comment by ranshid** [comment]
@naglera Please just add a short description in the top comment so that we do not have a link to a github comment in the git log.

**Pull request merged**